src/plugins/intel_gpu/src/plugin/graph.cpp

```diff
@@ -133,6 +133,23 @@ std::shared_ptr<cldnn::network> Graph::BuildNetwork(std::shared_ptr<cldnn::progr
     return network;
 }
 
+Graph::memory_states_map Graph::AllocateMemoryStates() {
+    Graph::memory_states_map states {};
+    const auto& memStatesInfo = m_program->GetMemoryStatesInfo();
+    for (const auto& memStateInfo : memStatesInfo) {
+        std::vector<cldnn::layout> orderedLayouts {memStateInfo.second.begin(), memStateInfo.second.end()};
+        std::sort(orderedLayouts.begin(), orderedLayouts.end(), [](cldnn::layout& first, cldnn::layout& second) {
+            return first.size.batch[0] < second.size.batch[0];
+        });
+        std::vector<cldnn::network::MemoryState::Ptr> memoryStates;
+        memoryStates.reserve(orderedLayouts.size());
+        for (const auto& layout : orderedLayouts)
+            memoryStates.push_back(std::make_shared<cldnn::network::MemoryState>(GetEngine()->allocate_memory(layout)));
+        states.insert({memStateInfo.first, memoryStates });
+    }
```

Comment:
Simple curiosity.
Why not allocating these varaibles memory at cldnn::network creation as allocate_outputs or allocate_intermediate buffers are done?